src/main/java/game_ui/GameplayScreen.java

```diff
@@ -0,0 +1,73 @@
+package game_ui;
+
+import interface_adapters.controllers.GameStartController;
+import interface_adapters.view_models.GameplayScreenModel;
+import interface_adapters.view_models.PlayerDataModel;
+import interface_adapters.view_models.ScreenUpdateListener;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class GameplayScreen extends JPanel implements ActionListener, ScreenUpdateListener {
+
+    private GameplayScreenModel gameplayScreenModel;
+    private GameStartController gameStartController;
+
+    public GameplayScreen(GameplayScreenModel gameplayScreenModel, GameStartController gameStartController) {
+        this.gameplayScreenModel = gameplayScreenModel;
+        this.gameStartController = gameStartController;
+    }
+
+    public void updateGameplayScreen() {
+        this.setLayout(new GridBagLayout());
+
+        PlayerDataModel playerData1 = gameplayScreenModel.getPlayer1();
+        PlayerDataModel playerData2 = gameplayScreenModel.getPlayer2();
+
+        CardButton p1creature1 = new CardButton(playerData1.getCreatureIds().get(0), playerData1.getCreatureNames().get(0));
+        CardButton p1creature2 = new CardButton(playerData1.getCreatureIds().get(1), playerData1.getCreatureNames().get(1));
+        CardButton p1creature3 = new CardButton(playerData1.getCreatureIds().get(2), playerData1.getCreatureNames().get(2));
+
+        CardButton p2creature1 = new CardButton(playerData2.getCreatureIds().get(0), playerData2.getCreatureNames().get(0));
+        CardButton p2creature2 = new CardButton(playerData2.getCreatureIds().get(1), playerData2.getCreatureNames().get(1));
+        CardButton p2creature3 = new CardButton(playerData2.getCreatureIds().get(2), playerData2.getCreatureNames().get(2));
+
+        this.add(p1creature1, getGBC(3, 6, 1, 1, 0, 0, 2, 3));
+        this.add(p1creature2, getGBC(5, 6, 1, 1, 0, 0, 2, 3));
+        this.add(p1creature3, getGBC(7, 6, 1, 1, 0, 0, 2, 3));
+        this.add(p2creature1, getGBC(3, 2, 1, 1, 0, 0, 2, 3));
+        this.add(p2creature2, getGBC(5, 2, 1, 1, 0, 0, 2, 3));
+        this.add(p2creature3, getGBC(7, 2, 1, 1, 0, 0, 2, 3));
```

Comment:
Obviously doesn't make things much simpler but allows for scalability and maybe some inspiration for if we wanted to add more creatures/players, just could loop it

```
for (int i = 0; i < 3; i++) {
    String creatureId = playerData1.getCreatureIds().get(i);
    String creatureName = playerData1.getCreatureNames().get(i);
    CardButton p1Creature = new CardButton(creatureId, creatureName);
    player1Buttons.add(p1Creature);
}

// Loop to create CardButton objects for player 2
for (int i = 0; i < 3; i++) {
    String creatureId = playerData2.getCreatureIds().get(i);
    String creatureName = playerData2.getCreatureNames().get(i);
    CardButton p2Creature = new CardButton(creatureId, creatureName);
    player2Buttons.add(p2Creature);
}

// Add the CardButton objects to the container using loop
for (int i = 0; i < 3; i++) {
    this.add(player1Buttons.get(i), getGBC(3 + i * 2, 6, 1, 1, 0, 0, 2, 3));
    this.add(player2Buttons.get(i), getGBC(3 + i * 2, 2, 1, 1, 0, 0, 2, 3));
}

Reply:
sure let's use this instead, will be good for adding listeners too

Reply:
oh this isnt part of a code suggestion, ok we'll replace what we have with this in the working branch